src/waveform/renderers/allshader/waveformrendererendoftrack.cpp

```diff
@@ -79,10 +80,8 @@ bool WaveformRendererEndOfTrack::preprocessInner() {
             kBlinkingPeriodMillis;
 
     const double remainingTime = m_pTimeRemainingControl->get();
-    const double remainingTimeTriggerSeconds =
-            WaveformWidgetFactory::instance()->getEndOfTrackWarningTime();
-    const double criticalIntensity = (remainingTimeTriggerSeconds - remainingTime) /
-            remainingTimeTriggerSeconds;
+    const double criticalIntensity = (m_remainingTimeTriggerSeconds - remainingTime) /
+            m_remainingTimeTriggerSeconds;
```

Comment:
Isn't this configurable from the user preferences? This would ignore that user preference. Should the skin designer have to wire that up?

Reply:
Yes, that's right. The reason for doing so is to allow renderer to be used in some other context (e.g S4 screen rendering), where different setting may be taken into account

Reply:
Right, but that means the user preference is being ignored by default. So this seems like a regression to me.

Reply:
Good catch - I fixed it in bffc6f335389d4131233ca8fba4e29d1393e97f5. I tested that it would dynamically update when editing the settings (since the singleton wouldn't be used in place) and it works well. This allows not needing references to that bloated singleton on the QML code (for which this renderer code is also used)